### PR TITLE
refactor: replace Reconcile with client-go

### LINKED ISSUE #4: [feature request] Refactor: replace `Reconcile`  with `client-go`
I think `client-go` informer style is more appropriate than `Reconcile` style  in `raven-agent` controller.
https://github.com/openyurtio/raven/blob/6b14ecb13076bf2bfd61580b4313169ba5c8850e/pkg/k8s/handlers.go#L41

I suggest we should replace the `Reconcile` style with `client-go` style.

/kind feature

### PR DESCRIPTION
**Pull request by DrmagicE**
fix #4

### CODE REVIEW

File: pkg/k8s/engine_controller.go [hunk 1]
```diff
@@ -108,33 +149,49 @@ func (c *EngineController) shouldHandleGateway(gateway *v1alpha1.Gateway) bool {
...
+func (c *EngineController) addGateway(obj interface{}) {
 	gw := obj.(*v1alpha1.Gateway)
 	if !c.shouldHandleGateway(gw) {
 		klog.InfoS("skip handle create gateway", "gateway", gw.Name, "node", c.nodeName)
 		return
 	}
-	klog.Info("handle create gateway: ", gw.Name)
-	c.handleCreateOrUpdateGateway(gw)
+	c.enqueue(gw, GatewayAdd)
+	return
```
Comment (BSWANG) [inline pkg/k8s/engine_controller.go:159]: unnecessary return statement

File: pkg/k8s/engine_controller.go [hunk 2]
```diff
@@ -108,33 +149,49 @@ func (c *EngineController) shouldHandleGateway(gateway *v1alpha1.Gateway) bool {
...
 
-func (c *EngineController) handleDeleteGateway(obj interface{}) {
+func (c *EngineController) deleteGateway(obj interface{}) {
 	gw := obj.(*v1alpha1.Gateway)
 	if !c.shouldHandleGateway(gw) {
 		klog.InfoS("skip handle delete gateway", "gateway", gw.Name, "node", c.nodeName)
 		return
 	}
+	c.enqueue(gw, GatewayDelete)
+	return
```
Comment (BSWANG) [inline pkg/k8s/engine_controller.go:179]: unnecessary return statement

**Review by BSWANG — CHANGES_REQUESTED** [review]

**Comment by njucjc** [comment]
lgtm

File: pkg/k8s/engine_controller.go (hunk 2 above)
Comment (DrmagicE) [reply pkg/k8s/engine_controller.go]: fixed

File: pkg/k8s/engine_controller.go (hunk 1 above)
Comment (DrmagicE) [reply pkg/k8s/engine_controller.go]: fixed

**Pull request merged**